src/store/aggregator/sagas/transferTokens.test.ts

```diff
@@ -0,0 +1,347 @@
+import { expectSaga } from 'redux-saga-test-plan';
+import {
+  EffectProviders,
+  StaticProvider,
+} from 'redux-saga-test-plan/providers';
+import { TransactionReceipt } from '@ethersproject/providers';
+
+import { parseUnits } from 'ethers/lib/utils';
+
+import {
+  accountSelector,
+  massetContractSelector,
+} from '../../app/app.selectors';
+import { aggregatorActions, aggregatorReducer } from '../aggregator.slice';
+
+import { transferTokens } from './transferTokens';
+import { AggregatorState } from '../aggregator.state';
+import {
+  depositMockValues,
+  mockAccount,
+  mockBassetAddress,
+  mockBridge,
+  mockMasset,
+  mockMassetAddress,
+  mockToken,
+  mockTokenAddress,
+  mockTokenDecimals,
+} from '../aggregator.mock';
+import {
+  bassetAddressSelector,
+  bridgeContractSelector,
+  flowStateSelector,
+  massetAddressSelector,
+  startingTokenAddressSelector,
+  startingTokenContractSelector,
+  startingTokenDecimalsSelector,
+} from '../aggregator.selectors';
+
+const aggregatorInitialState: AggregatorState = {
+  ...new AggregatorState(),
+};
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('transferTokens', () => {
+  const mockExtendTx = {
+    hash: '0x01',
+    wait: jest.fn() as ContractTransaction['wait'],
+  } as ContractTransaction;
+
+  const mockExtendReceipt = {
+    transactionHash: '0x01',
+  } as TransactionReceipt;
+
+  const initialSteps = aggregatorInitialState?.submitCall?.steps;
+
+  describe('deposit', () => {
+    const mockSelectors: (EffectProviders | StaticProvider)[] = [
+      [matchers.select(flowStateSelector), 'deposit'],
+      [matchers.select(bridgeContractSelector), mockBridge],
+      [matchers.select(startingTokenAddressSelector), mockTokenAddress],
+      [matchers.select(startingTokenDecimalsSelector), mockTokenDecimals],
+      [matchers.select(startingTokenContractSelector), mockToken],
+      [matchers.select(massetAddressSelector), mockMassetAddress],
+      [matchers.select(accountSelector), mockAccount],
+    ];
+
+    it('when sufficient allowance', async () => {
+      const expectedState: AggregatorState = {
+        ...aggregatorInitialState,
+        submitCall: {
+          status: 'success',
+          currentOperation: 'deposit',
+          steps: [
+            {
+              ...initialSteps?.[2],
+              tx: mockExtendTx,
+              txReceipt: mockExtendReceipt,
+            },
+          ],
+        },
+      };
+
+      await expectSaga(
+        transferTokens,
+        aggregatorActions.submit(depositMockValues)
+      )
+        .withReducer(aggregatorReducer)
+        .withState(aggregatorInitialState)
+        .provide([
+          ...mockSelectors,
+          [matchers.call.fn(mockToken.allowance), parseUnits('11')],
+          [matchers.call.fn(mockBridge.receiveTokensAt), mockExtendTx],
+          [matchers.call.fn(mockExtendTx.wait), mockExtendReceipt],
+        ])
+        .hasFinalState(expectedState)
```

Comment:
Add `.call` checks for `mockToken.allowance` and `mockBridge.receiveTokensAt` to verify that contract methods are called with proper arguments